packages/core/contracts/financial-templates/contract-for-difference/ContractForDifference.sol

```diff
@@ -145,6 +149,7 @@ contract ContractForDifference is Testable, Lockable {
             "Ancillary Data too long"
         );
         customAncillaryData = _customAncillaryData;
+        prepaidProposerReward = _prepaidProposerReward;
```

Comment:
if this contract is created directly (not via the factory) it is possible to make settlement revert if the LSP does not have the balance of the reward to pay on the `_requestOraclePriceExpiration()` method.

I don't really think this is an issue because:
1) people will use the factor almost always
2) the contract is not broken in this case as an external EOA can always send enough tokens to the contract to let settlement continue.

Reply:
Yes, yes, and yes :).

Those were my exact thoughts on it. I thought about trying to predict the address and pre-approve the tokens, but the code to predict deployment addresses in solidity is complex. And people should always use the factory. We have a similar issue if someone were to create an EMP without a factory. it would look fine, but when a dispute gets sent, it fails. I think we can make the same assumption here.